src/mscorlib/src/System/IO/MemoryStream.cs

```diff
@@ -391,6 +391,36 @@ public override int Read([In, Out] byte[] buffer, int offset, int count)
             return n;
         }
 
+        public override int Read(Span<byte> destination)
+        {
+            if (GetType() != typeof(MemoryStream))
+            {
+                // MemoryStream is not sealed, and a derived type may have overridden Read(byte[], int, int) prior
+                // to this Read(Span<byte>) overload being introduced.  In that case, this Read(Span<byte>) overload
+                // should use the behavior of Read(byte[],int,int) overload.
+                return base.Read(destination);
+            }
+
+            if (!_isOpen)
+            {
+                __Error.StreamIsClosed();
+            }
+
+            int n = Math.Min(_length - _position, destination.Length);
+            if (n <= 0)
+            {
+                return 0;
+            }
+
+            // TODO: Read(byte[], int, int) has an n <= 8 optimization, presumably
```

Comment:
It's effectively tracked by https://github.com/dotnet/corefx/issues/22388, as completing that issue includes making sure it's implemented well.  I'll add that issue number to the comment.